geetools/ee_feature_collection.py

```diff
@@ -504,3 +506,51 @@ def plot(
             gdf.boundary.plot(ax=ax, color=color)
         else:
             gdf.plot(column=property, ax=ax, cmap=cmap)
+
+    @classmethod
+    def fromGeoInterface(cls, data: Any) -> ee.FeatureCollection:
```

Comment:
This might be a good use case for [`subtyping`](https://peps.python.org/pep-0544/) with `typing.Protocol`. It would allow typecheckers and IDE to check for the presence of __geo_interface__ before runtime.

Base on [GeoPandas implementation](https://github.com/geopandas/geopandas/blob/main/geopandas/geodataframe.py#L964-L996) it could look like this:

```python
from typing import Protocol

class GeoInterface(Protocol):
    @property
    def __geo_interface__(self) -> dict:
        ...
```

Reply:
implemented in https://github.com/gee-community/geetools/pull/381/commits/5deeb6d4bebdbded357bce9d26e1c4539ad16ddc